src/pages/CartPage.ts

```diff
@@ -1,59 +1,57 @@
 import { expect, test, type Locator, type Page } from '@playwright/test';
-import { AbstractPage } from './BasePage';
+import { BasePage } from './BasePage';
  
-export class CartPage extends AbstractPage {
-  readonly page: Page;
+export class CartPage extends BasePage {
   readonly emptyCartMessage: Locator;
+  readonly url: string = '/cart';
+  readonly name: string;
 
   constructor(page: Page) {
     super(page);
-    this.page = page;
-    this.url = './cart';
+    this.name = 'Cart';
```

Comment:
This is redundant if we add pageName() method to the BasePage.

Reply:
fixed